cmd/chart.go

```diff
@@ -25,142 +22,99 @@ const (
+	cmd := &cobra.Command{
+		Use:     "move <chart>",
+		Short:   "Relocates a Helm Chart along with their associated container images",
+		Long:    "It takes the provided Helm Chart, resolves and repushes all the dependent images, providing as output a modified Helm Chart (and subcharts) pointing to the new location of the images.",
+		Example: "move my-chart --image-patterns my-image-hints.yaml --registry my-registry.company.com ",
+		RunE:    MoveChart,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return errors.New("requires a chart argument")
+			}
+			return nil
+		},
 	}
 
-	sourceChart, err := loader.Load(args[0])
-	if err != nil {
-		return nil, fmt.Errorf("failed to load helm chart at \"%s\": %w", args[0], err)
-	}
+	f := cmd.Flags()
+	// TODO(miguel): Change to image-hints
+	f.StringVarP(&imagePatternsFile, "image-patterns", "i", "", "dile with image patterns")
+	f.BoolVarP(&skipConfirmation, "yes", "y", false, "Proceed without prompting for confirmation")
 
-	return sourceChart, nil
-}
+	f.StringVar(&registryRule, "registry", "", "hostname of the registry used to push the new images")
+	f.StringVar(&repositoryPrefixRule, "repo-prefix", "", "path prefix to be used when relocating the container images")
 
-func loadImagePatterns(chart *chart.Chart) (string, error) {
-	patterns, err := mover.LoadImagePatterns(ImagePatternsFile, chart)
-	if err != nil {
-		return "", fmt.Errorf("failed to read image pattern file: %w", err)
-	}
-	if patterns == "" {
-		return patterns, fmt.Errorf("image patterns file is required. Please try again with '--image-patterns <image patterns file>'")
-	}
-	if ImagePatternsFile == "" {
-		log.Println("Using embedded image patterns file.")
-	}
-	return patterns, nil
-}
+	f.UintVar(&retries, "retries", defaultRetries, "number of times to retry push operations")
+	f.StringVar(&output, "out", "./*.relocated.tgz", "output chart name produced")
 
-func loadRules() (*rewrite.Rules, error) {
-	rules := &rewrite.Rules{}
-	if RulesFile != "" {
-		var err error
-		rules, err = rewrite.ParseRules(RulesFile)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if RegistryRule != "" {
-		rules.Registry = RegistryRule
-	}
-
-	if RepositoryPrefixRule != "" {
-		rules.RepositoryPrefix = RepositoryPrefixRule
-	}
-
-	return rules, nil
+	return cmd
 }
 
 func MoveChart(cmd *cobra.Command, args []string) error {
-	sourceChart, err := loadChartFromArgs(args)
-	if err != nil {
-		return err
+	if registryRule == "" && repositoryPrefixRule == "" {
+		return fmt.Errorf("at least one rewrite rule must be given. Please try again with --registry and/or --repo-prefix")
 	}
 
-	imagePatterns, err := loadImagePatterns(sourceChart)
-	if err != nil {
-		return err
+	targetRewriteRules := &mover.OCIImageRewriteRules{
+		Registry:         registryRule,
+		RepositoryPrefix: repositoryPrefixRule,
 	}
 
-	rules, err := loadRules()
+	chartMover, err := mover.NewChartMover(
+		args[0],
```

Comment:
How are we sure this will be always correct?
what if they put the flags before the chart filename? does cobra eat those up from args?

Reply:
Cobra handles that as expected, I also added the validation as part of the command description to ensue there is at least one entry in the args slice https://github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/pull/37/files#diff-b40d51eb9f0228b78570a572c410b9016ee9f3d9b8ee24b4e6a43fc6e5408db0R57